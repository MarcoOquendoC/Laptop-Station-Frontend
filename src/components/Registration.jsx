import React from "react";
import { Link, Navigate } from 'react-router-dom';

const Registration = () => {


  return (
    <>
      <div>
        <form action="">
          <input type='text' />
          <input type='text' />
          <input type='text' />
          <input type='text' />
          <input type='text' />
          <button type="button">sign in</button>
        </form>
      </div>
    </>
  );
};

export default Registration;