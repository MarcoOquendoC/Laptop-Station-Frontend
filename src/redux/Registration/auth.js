import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
import api from '../../api/api';

// Action creators
const SIGNUP = 'SIGNUP';
const LOGIN = 'LOGIN';
const LOGOUT = 'LOGOUT';
const GET_AUTH_USER = 'GET_AUTH_USER';

const initialState = [];

// Thunk
const register = createAsyncThunk(SIGNUP, async (user) => {
  try {
    return await api.register(user);
  } catch (error) {
    return error.message;
  }
});

const login = createAsyncThunk(LOGIN, async (user) => {
  try {
    return await api.login(user);
  } catch (error) {
    return error.message;
  }
});

const logout = createAsyncThunk(LOGOUT, async () => {
  try {
    return await api.logout();
  } catch (error) {
    return error.message;
  }
});

const currentUser = createAsyncThunk(GET_AUTH_USER, async () => {
  try {
    return await api.currentUser();
  } catch (error) {
    return error.message;
  }
});

// Reducer
const authSlice = createSlice({
  name: 'auth',
  initialState,
  reducers: { },
  extraReducers: {
    [register.fulfilled]: (state, action) => action.payload,
    [login.fulfilled]: (state, action) => action.payload,
    [logout.fulfilled]: (state, action) => action.payload,
    [currentUser.fulfilled]: (state, action) => action.payload,
  },
});

export {
  register, login, logout, currentUser,
};
export default authSlice.reducer;
