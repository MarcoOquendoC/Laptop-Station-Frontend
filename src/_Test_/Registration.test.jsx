import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import Registration from '../components/Registration';
import store from '../redux/configureStore';

describe('Testing Home', () => {
  let signup;
  beforeAll(async () => {
    signup = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Registration />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(signup).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(signup).toMatchSnapshot();
  });

  it('renders the first name input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('first name'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('first name'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('renders the last name input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('last name'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('last name'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('renders the email input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('email'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('email'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('renders the password input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('password'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('password'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });

  it('renders the confirm password input field', () => {
    const emailInput = waitFor(() => screen.getByPlaceholderText('confirm password'));
    waitFor(() => expect(emailInput).toBeInTheDocument());
    waitFor(() => expect(emailInput.type).toBe('confirm password'));
    waitFor(() => expect(emailInput.required).toBe(true));
  });
});
