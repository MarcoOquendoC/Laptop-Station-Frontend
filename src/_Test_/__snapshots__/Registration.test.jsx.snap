// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Home should match the snapshot 1`] = `
Array [
  <div
    className="principal"
  >
    <div
      className="header"
    >
      <div
        className="menu mob"
      >
        <img
          alt="menu"
          src="menu.svg"
        />
      </div>
      <div
        className="menu_logo"
      >
        <img
          alt="logo"
          className="logo"
          src="logo_laptop_station.svg"
        />
      </div>
      <div
        className="menu mob"
      />
    </div>
    <nav
      className="menunav"
    >
      <ul
        className="menunav_ul"
      >
        <li
          className="menu_text"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            Log in
          </a>
        </li>
        <li
          className="menu_text select"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/laptops"
            onClick={[Function]}
          >
            Latops
          </a>
        </li>
        <li
          className="menu_text"
        >
          <a
            href="/reservations"
            onClick={[Function]}
          >
            Reserves
          </a>
        </li>
        <li
          className="menu_text admin"
        >
          <a
            className="a__admin"
            href="/addItem"
            onClick={[Function]}
          >
            Add item
          </a>
        </li>
        <li
          className="menu_text"
        />
      </ul>
    </nav>
  </div>,
  <div
    className="viewfinder"
  >
    <div
      className="register"
    >
      <h1>
        Sign In
      </h1>
      <form
        className="form"
        onSubmit={[Function]}
      >
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="first name"
          type="text"
        />
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="last name"
          type="text"
        />
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="email"
          type="email"
        />
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="password"
          type="password"
        />
        <input
          className="form__input"
          onChange={[Function]}
          placeholder="confirm password"
          type="password"
        />
        <div
          className="form__input"
        />
        <button
          className="form__btn"
          type="submit"
        >
          Sign up
        </button>
      </form>
      <p>
        Or
        <a
          className="orsignup"
          href="/login"
          onClick={[Function]}
        >
           Log in 
        </a>
        if you already have an account.
      </p>
    </div>
  </div>,
]
`;
